action/sealedenvelope.go

```diff
@@ -7,19 +7,39 @@ import (
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/pkg/errors"
 
+	"github.com/iotexproject/iotex-core/config"
 	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
 )
 
 // SealedEnvelope is a signed action envelope.
 type SealedEnvelope struct {
 	Envelope
+	encoding      uint32
+	externChainID uint32
```

Comment:
externChainID is not part of the struct, because it can be told from the encoding for now.

Reply:
we discussed last night, it cannot be told from the encoding (unless we send in all tx bytes, then you can decode and get it)

for now, we'll just set it according to your network config (4689/4690), and use it to calculate raw hash, and verify against the signature
if signature fail, means this tx is not intended for this network

Reply:
besides https://github.com/iotexproject/iotex-core/pull/2592/files#diff-fda8afdb7165ec30b8d2c40dfd1963def0ffeb117a3d9098167b8139c72527caR97, where else we set the externChainID? If we can use confg.ExternChainID() there for rlp encoding, why cannot we do it in the struct?

Reply:
that's the only place we set externChainID

what do you mean exactly "why cannot we do it in the struct?", as said above, that is the ONLY place where we need to set externChainID, since it only applies for IsRLP() case